projects/transformers/run.py

```diff
@@ -441,16 +471,28 @@ def model_init():
     # Update any extra kwargs defined in config
     hp_search_kwargs.update(**model_args.hp_extra_kwargs)
 
-    # Run hp search and save results
-    best_run = trainer.hyperparameter_search(**hp_search_kwargs)
-    logging.info(f"Best run: {best_run}")
+    # Run hp search and save results. Code to remove checkpoints won't get
+    # called if ANY of the trials error out, so wrap with try/except.
+
+    success = False
+    try:
+        best_run = trainer.hyperparameter_search(**hp_search_kwargs)
+        logging.info(f"Best run: {best_run}")
+        success = True
+    except TuneError:
+        logging.info(f"One or more trials errored out")
+    finally:
+        logging.info("An unknown error occured during hp search.")
```

Comment:
This probably doesn't belong in the finally block, but in the except block - the finally block will execute regardless where it fails or not. So even if it succeeds you will get an error message saying an unknown error occurred.

Finally is only useful if you wanna make sure things like saving X or closing Y is always done regardless of whether the code executes or not.

Reply:
Whoops, good catch here as well.

Reply:
In the updated code, the finally block is used to call the code to remove unnecessary checkpoints. That does need to run regardless of rather or not hp search completes.